packages/react-reconciler/src/ReactFiberHydrationContext.js

```diff
@@ -555,6 +629,27 @@ function prepareToHydrateHostTextInstance(fiber: Fiber): void {
   }
 }
 
+function prepareToHydrateHostActivityInstance(fiber: Fiber): void {
+  if (!supportsHydration) {
+    throw new Error(
+      'Expected prepareToHydrateHostActivityInstance() to never be called. ' +
+        'This error is likely caused by a bug in React. Please file an issue.',
+    );
+  }
+  const activityState: null | ActivityState = fiber.memoizedState;
+  const activityInstance: null | ActivityInstance =
+    activityState !== null ? activityState.dehydrated : null;
+
+  if (!activityInstance) {
```

Comment:
>_possible bug_

**✨ AI Suggestion:** The variable `activityInstance` is already checked for nullity in the outer `if` statement. The nested null check `!activityInstance` is redundant.
```suggestion
    const activityState: null | ActivityState = fiber.memoizedState;
    const activityInstance: null | ActivityInstance =
      activityState !== null ? activityState.dehydrated : null;

    if (activityInstance === null) {
```